[tools] Add basic fixing capabilities to the copyright linter tool

sgnn7 opened the pull request:
### What does this PR do?

Previously, the `inv lint-copyrights` tooling was only able to detect
problems with the copyright headers. This change adds functionality to
fix the vast majority of these issues (specifically when the header is
completely missing) when run with `--fix` flag. A `--dry-run` flag was
also added as a way to safely test run this command before making
permanent changes to files.

### Motivation

Automation of copyright header fixing

### Additional Notes

N/A

### Possible Drawbacks / Trade-offs

We currently can only handle basic cases of linting issues such
as a completely missing header but this covers >90% of current
linter hits.

### Describe how to test/QA your changes

```sh-session
$ inv lint-copyrights --fix

$ git diff
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

sgnn7 committed:
[tools] Add basic fixing capabilities to the copyright linter tool

Previously, the `inv lint-copyrights` tooling was only able to detect
problems with the copyright headers. This change adds functionality to
fix the vast majority of these issues (specifically when the header is
completely missing) when run with `--fix` flag. A `--dry-run` flag was
also added as a way to safely test run this command before making
permanent changes to files.

mx-psi reviewed (commented):
This is great, it would be awesome to have this as a precommit hook to automatically fix files

sgnn7 commented:
> it would be awesome to have this as a precommit hook to automatically fix files

👍 I am hoping to get it to that point :)

sgnn7 committed:
[tools] Make the copyright linter `--fix` support new Go build flags

This change ensures that we can handle also inserting the copyright
header into files that start with the new Golang build flag comments and
not just the older ones.

sgnn7 commented on tasks/libs/copyright.py:
```diff
@@ -153,15 +161,61 @@ def _assert_copyrights(files, debug=False):
+
+        # If the file starts with a comment and it's not a build comment,
+        # there is likely a manual fix to the header needed
+        if header[0].startswith("//") and not header[0].startswith("// +build "):
```
Fixed via https://github.com/DataDog/datadog-agent/pull/10159/commits/b0af052dd41407bfd8c384c2653a398e5df54c53

sgnn7 committed:
[tools] Add more header exclusion patterns to copyright header linter

There are a few more patterns to be added in the exclusion filters so
that they are not flaged as false postives for fixing by this linter.

sgnn7 committed:
[tools] Add '//+build' to fixable files for the copyright linter

This start of file is pretty much the same as the one with a space so
the fix of prepending the header should work fine on them.

sgnn7 committed:
[tools] Add auto-generated probe files to copyright linting exclusion filters

These files should not be touched as they are gneerated but they do not
have the headers to indicate it so we have to exclude them via paths.
